dataclasses.py

```diff
@@ -602,3 +603,72 @@ def _astuple_inner(obj, tuple_factory):
                           for k, v in obj.items())
     else:
         return deepcopy(obj)
+
+
+def replace(obj, **changes):
```

Comment:
I'm guessing a discussion we're going to have to have (or did we have it already?) is whether these are better as methods starting with an underscore (e.g. `obj._replace`). Then you don't need the `isdataclass()` check. But you impose on the user's namespace more. But it would be more like namedtuple, which has `_asdict()` and `_replace()` as methods. (FWIW I'm undecided, but seeing the `isdataclass()` check here makes me think that these really *want* to be methods. Don't let this stop you from merging this though.)

Reply:
Yeah, we had this discussion in #8. But the more I use it, the more I've been thinking that the module-level functions are clumsy.